cmd/agent/api/server.go

```diff
@@ -129,6 +129,10 @@ func StartServer() error {
 		// seconds, need to find a solution for that before
 		// re-enabling.
 		// WriteTimeout: config.Datadog.GetDuration("server_timeout") * time.Second,
+		ConnContext: func(ctx context.Context, c net.Conn) context.Context {
```

Comment:
NOTE: I implemented this before the above `WriteTimeout` was disabled. This change probably fixes the issue outlined in the above TODO. But so long as `WriteTimeout` is disabled, the work in this PR to override the read/write timeout per request won't really do anything (since it's already disabled)